### PR TITLE
feat(event-receivers): add version targeting

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes

- Add version targeting for event dispatcher receiver version
- Fix deployed helm release version performing checkout against release version before helm upgrade command

#### Motivation and Context
This modification is needed in order to target a specific version (PROD/STAGING) for event dispatcher when sending event receiver command such as START/STOP.
This is required to achieve zero down-time deploy when a new app version is add during deploy and those commands are used to selectively turn event receivers down based on version.

For the helm release fix: actually deploy is performed with a self checkout that points to the github commit ref for the runned pipeline. 
During pipeline execution version bump phase is executed that create a release and a tag on github based on the specified version bump but the github refs points to the commit before that and this result in helm release being performed with the chart and app version valued with the values before version bump

#### How Has This Been Tested?
Junit tests + local tests with eCommerce local

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: helm/values-prod.yaml [hunk 1]
```diff
@@ -173,6 +173,7 @@ microservice-chart:
     NODE_FORWARDER_READ_TIMEOUT: "10000"
     NODE_FORWARDER_CONNECTION_TIMEOUT: "10000"
     REDIRECT_PAYMENT_TYPE_CODES: "PPAYITR1XXX-RBPB,CHECKOUT-PPAYITR1XXX-RBPP,PPAYITR1XXX-RBPR"
+    DEPLOYMENT_VERSION: ""
```
Comment (infantesimone) [inline helm/values-prod.yaml:176]: if the string is empy, you consider PROD as default in application.properties, correct? works?

**Review by infantesimone — DISMISSED** [review]
really good, just some consideration

File: helm/values-prod.yaml (hunk 1 above)
Comment (pietro-tota) [reply helm/values-prod.yaml]: Good catch, no, an empty string would be used here, fixed with https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/322/commits/567a7bbc7bfd9205986d503dff973a6bbc12e6f6 to set it explicitly to PROD.

**Comment by pietro-tota** [comment]
Note for reviewers (and reminder for me :D):

As deploy strategy this PR have to be merged and deployed first, adding per deployment version receivers stop capability.
This PR introduce a breaking change in redis stream events that now expects a target field.
This is not a problem, when this version will be deployed the redis stream stop event (that will be the latest written in redis stream before starting deploy in order to stop receivers) will generate a parsing error so newly deployed instance receivers will stay up.

This will be a controlled error that will cause no "damage" even in case of rollback: the new stop event will be "ignored" by previous version, with event receivers that will not be stopped once old version is restored.

Any stop/start can be performed by the currently installed version using the stop/start apis as done per others deployment till now.

**Review by infantesimone — APPROVED** [review]

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -94,6 +94,7 @@ redisStream.eventController.consumerGroupPrefix=${REDIS_STREAM_EVENT_CONTROLLER_
 redisStream.eventController.consumerNamePrefix=${REDIS_STREAM_EVENT_CONTROLLER_CONSUMER_NAME_PREFIX}
 redisStream.eventController.failOnErrorCreatingConsumerGroup=${REDIS_STREAM_EVENT_CONTROLLER_FAIL_ON_ERROR_CREATING_CONSUMER_GROUP:true}
 eventController.status.pollingChron=${EVENT_CONTROLLER_STATUS_POLLING_CHRON}
+eventController.deploymentVersion=${DEPLOYMENT_VERSION:PROD}
```
Comment (CianoDanilo) [inline src/main/resources/application.properties:97]: Is it correct to use PROD as default value also for dev and uat environments?

**Review by CianoDanilo — APPROVED** [review]

**Pull request merged**